python/vineyard/deploy/local.py

```diff
@@ -148,4 +156,86 @@ def start_vineyardd(etcd_endpoints=None,
             etcd_ctx.__exit__(None, None, None)  # pylint: disable=no-member
 
 
+def init(proc_num=1, socket=None, **kw):
+    '''
+    Help new users to launch a local vineyardd and get a client as easy as possible
+
+    In a clean enviroment, simply use:
+    
+    .. code::
+    
+        vineyard.init()
+    
+    it will launch a local vineyardd and return a connected client to the vineyardd.
+
+    If a specific **SOCKET** path is needed, pass the socket either by setting the
+    environment variable **VINEYARD_IPC_SOCKET** as the SOCKET before head, or simply use:
+    
+    .. code::
+    
+        vineyard.init(socket=SOCKET)
+
+    For the case to establish a local vineyard cluster consists of multiple vineyard daemons.
+    Use the **proc_num** parameter:
+
+    .. code::
+    
+        client1, client2, client3 = vineyard.init(proc_num=3)
+    
+    A socket prefix can also be provided if needed:
+
+    .. code::
+
+        client1, client2, client3 = vineyard.init(socket='/tmp/vineyard.sock', proc_num=3)
+
+    In this case, three vineyard daemons at **/tmp/vineyard.sock.0**, **/tmp/vineyard.sock.1**
+    and **/tmp/vineyard.sock.2** will be launched.
+
+    The init method can only be called once in a process, to get the established clients later
+    in the process, use **get_default_client**.
+    '''
+    assert init.contexts == {}
+
+    if socket is None:
+        socket = os.environ.get('VINEYARD_IPC_SOCKET', None)
+
+    etcd_endpoints = None
+    etcd_prefix = f'vineyard_init_at_{time.time()}'
+    for idx in range(proc_num):
+        if proc_num > 1:
+            ipc_socket = f'{socket}.{idx}' if socket else None
+        else:
+            ipc_socket = socket
+        try:
+            client = connect(ipc_socket)
+        except:
+            logger.info("Vineyard daemon at %s is not initialized", ipc_socket)
+            ctx = start_vineyardd(socket=ipc_socket,
+                                  etcd_endpoints=etcd_endpoints,
+                                  etcd_prefix=etcd_prefix,
+                                  rpc=False,
+                                  **kw)
+            _, ipc_socket, etcd_endpoints = ctx.__enter__()
+            client = connect(ipc_socket)
+            init.contexts[ipc_socket] = (ctx, client)
+
+    return get_default_client()
+
+
+def get_default_client():
+    if not init.contexts:
+        raise ValueError("Vineyard has not been initialized, use vineyard.init()")
+    clients = [init.contexts[k][1] for k in init.contexts]
+    return clients if len(clients) > 1 else clients[0]
+
+
+@atexit.register
+def shutdown():
+    for ctx in reversed(init.contexts.keys()):
+        init.contexts[ctx][0].__exit__(None, None, None)
+    init.contexts = {}
```

Comment:
Please do something like

```python
def shutdown()....

@atexit.register
def __shutdown_handler():
     try:
          shutdown()
     except Exception:
          pass
```

to avoid exceptions in `shutdown()` when `atexit` breaking other parts of user's program.